src/main/java/omero/gateway/util/Links.java

```diff
@@ -0,0 +1,83 @@
+package omero.gateway.util;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import omero.gateway.model.DataObject;
+import omero.model.IObject;
+
+import org.apache.commons.beanutils.PropertyUtils;
+
+/**
+ * Some helper methods dealing with XXXYYYLinkI IObjects, e.g.
+ * ImageAnnotationLinkI
+ *
+ * @author Dominik Lindner &nbsp;&nbsp;&nbsp;&nbsp;
+ * <a href="mailto:d.lindner@dundee.ac.uk">d.lindner@dundee.ac.uk</a>
+ */
+public class Links {
+
+    private Links() {
+    }
+
+    /**
+     * Get the Link class for a certain child parent combination, e.g. parent:
+     * DatasetData, child: ImageData => omero.model.DatasetImageLinkI
+     *
+     * @param parent The parent
+     * @param child  The child
+     * @return See above
+     * @throws ClassNotFoundException
+     */
+    public static Class<? extends IObject> getLinkClass(
+            Class<? extends DataObject> parent,
+            Class<? extends DataObject> child) throws ClassNotFoundException {
+        String parentType = getType(parent);
+        String childType = getType(child);
+        String type = parentType + childType + "LinkI";
+        return (Class<? extends IObject>) Class.forName("omero.model." + type);
+    }
+
+    /**
+     * Set the parent, child or both of an existing link
+     *
+     * @param link   The link object
+     * @param parent The parent (or <code>null</code>)
+     * @param child  The child (or <code>null</code>)
+     * @return The link object
+     * @throws ClassNotFoundException
```

Comment:
Possibly not worth these throws lines if there's no actual documentation! Also may want to consider just brushing under the `ReflectiveOperationException` superclass.